src/customer/strategies/googlepay/googlepay-authorizenet-customer-strategy.spec.ts

```diff
@@ -0,0 +1,258 @@
+
+        jest.spyOn(formPoster, 'postForm')
+            .mockReturnValue(Promise.resolve());
+
+        container = document.createElement('div');
+        container.setAttribute('id', 'googlePayCheckoutButton');
+        walletButton = document.createElement('a');
+        walletButton.setAttribute('id', 'mockButton');
+
+        jest.spyOn(paymentProcessor, 'createButton')
+            .mockReturnValue(walletButton);
+
+        container.appendChild(walletButton);
+        document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+    });
+
+    it('creates an instance of GooglePayCustomerStrategy', () => {
+        expect(strategy).toBeInstanceOf(GooglePayCustomerStrategy);
+    });
+
+    describe('#initialize()', () => {
+
+        describe('Payment method exist', () => {
+
+            it('Creates the button', async () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+
+                await strategy.initialize(customerInitializeOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('fails to initialize the strategy if no GooglePayCustomerInitializeOptions is provided ', () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions(Mode.Incomplete);
+
+                function initialize() {
+                    strategy.initialize(customerInitializeOptions);
+                }
+
+                expect(initialize).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no methodid is supplied', () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions(Mode.UndefinedMethodId);
+
+                function initialize() {
+                    strategy.initialize(customerInitializeOptions);
+                }
+
+                expect(initialize).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no valid container id is supplied', () => {
+                customerInitializeOptions = getAuthNetCustomerInitializeOptions(Mode.InvalidContainer);
+
+                return expect(strategy.initialize(customerInitializeOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        let customerInitializeOptions: CustomerInitializeOptions;
+
+        beforeEach(() => {
+            customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+        });
+
+        it('succesfully deinitializes the strategy', async () => {
+            await strategy.initialize(customerInitializeOptions);
+
+            strategy.deinitialize();
+
+            if (customerInitializeOptions.googlepayauthorizenet) {
+                const button = document.getElementById(customerInitializeOptions.googlepayauthorizenet.container);
+
+                if (button) {
```

Comment:
having `if` conditions  inside a test is not desirable as it adds complexity. I think it's safe to assume they wiil always be present.

```js
const button = document.getElementById(customerInitializeOptions.googlepayauthorizenet!.container);
expect(button!.firstChild).toBe(null);
```

You can suppress the linter with an annotation

Reply:
Sorry, I forgot to refactor that part.